### PR TITLE
Fixes editing and pinning a post

### PR DESCRIPTION
**Pull request by koox00**
#### Summary

PR https://github.com/mattermost/mattermost-server/pull/19404 fixed replies on posts by casting StringArray to a string.

This PR fixes editing and pinning a post by casting `StringMap`, and
`StringInterface` to a string as well.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-41066

#### Release Note

```release-note
Fixed inserting a `jsonb` field into the DB when `binary_parameters=yes`
```

### CODE REVIEW

**Review by cpoile — APPROVED** [review]

File: model/utils.go [hunk 1]
```diff
@@ -124,7 +124,12 @@ func (m *StringMap) Scan(value interface{}) error {
 
 // Value converts StringMap to database value
 func (m StringMap) Value() (driver.Value, error) {
-	return json.Marshal(m)
+	j, err := json.Marshal(m)
+	if err != nil {
+		return nil, err
+	}
+	// non utf8 characters are not supported
```
Comment (Willyfrog) [inline model/utils.go:131]: In the comment I'd add a link to the ticket so people know why we are doing this.

**Review by Willyfrog — APPROVED** [review]
minor nit, but LGTM! thanks for the quick fix

**Comment by amyblais** [comment]
@koox00 There are some checks failing.

**Comment by koox00** [comment]
@amyblais there is an outage currently at circleci https://status.circleci.com/#

**Comment by jgilliam17** [comment]
@amyblais I started [E2Es here](https://automation-dashboard.vercel.app/cycles/5cb34b91-9736-48e9-beee-d0cd887ff8e9), despite build not being green. Will check shortly for failures, if the run ends up being successful.

**Comment by jgilliam17** [comment]
E2E report looks good https://mattermost-cypress-report.s3.amazonaws.com/108753-466ff0b-onprem-ent-server-pr-19425-mm-ee-test:466ff0b/mochawesome.html
I am creating a test server to test the changes manually, and check for MM-T247 that failed on the PR, but not on master.

**Comment by koox00** [comment]
we'll need circle to work though, so we see if the unit tests pass as well

**Comment by jgilliam17** [comment]
- Tested replying, editing and pinning a post - all 3 actions work as expected on the test server for the PR.
- MM-T247 passed when testing manually - @koox00 can you check if failure is related to PR changes? Thanks

**Review by agnivade — APPROVED** [review]
A test would be good for this. But what we really need is an e2e test in binary mode. Let me talk with Saturn to change our e2e framework.

**Comment by agnivade** [comment]
Actually I just sent https://github.com/mattermost/mattermost-server/pull/19432. Can you check if there's a store test we can write that fails in binary mode without this PR? Then just adding that PR should be enough.

**Comment by saturninoabril** [comment]
@agnivade Thanks for submitting MR to E2E test enable binary_parameters for postgres. It's merged and will run test with such change.

**Review by saturninoabril — APPROVED** [review]
@jgilliam17 @koox00 I ran again the [E2E](https://mattermost-cypress-report.s3.amazonaws.com/108874-fe0342b-onprem-ent-server-pr-19425-mm-ee-test:fe0342b/mochawesome.html), and all passed as expected, including MM-T247.

**Pull request merged**